### PR TITLE
Platform & MEC Service API Metrics

### PR DESCRIPTION
**Pull request by dilallkx**
**CHANGES:**
- Dashboard updates:
  - New Platform Metrics Dashboard with Service API Request & Notification metrics
  - Renamed _Node Exporter Full_ dashboard to _Runtime Environment Metrics (Node)_
- Bumped AdvantEDGE version to 1.6.6
- Instrumented Prometheus metrics to HTTP Requests & Notifications in all Platform & MEC Services
  - Added a ServiceMonitor resource to inform prometheus to start scraping services
  - Added a metrics handler on port 9000 for all platform microservices requiring metrics scraping
    - Default 10s scraping for all services 
- Renamed _meep-metric-store_ package to _meep-metrics_
  - Added MetricsHandler functionality to _meep-metrics_ package
- Auth Service: Cleaned up unused prometheus metrics
- Platform & MEC Services: Added metrics for notifications
- meepctl:
  - Disabled CRD deletion
  - Removing CRDs prevents UT execution without deploy dependency pods
- Frontend: Added links to new dashboards

**TESTING:**
- Cypress, UT & System Tests pass
- Manual verification of new dashboard integration

### CODE REVIEW

File: charts/meep-gis-engine/templates/service.yaml [hunk 1]
```diff
@@ -14,8 +14,15 @@ spec:
...
       nodePort: {{ .Values.service.nodePort }}
-      {{- end }}
\ No newline at end of file
+      {{- end }}
+    {{- if .Values.prometheus.monitor.enabled}}
+    - name: metrics
+      port: {{ .Values.prometheus.monitor.port }}
+      targetPort: {{ .Values.prometheus.monitor.port }}
+      protocol: TCP
+    {{- end}}
```
Comment (roymx) [inline charts/meep-gis-engine/templates/service.yaml:28]: Missing newline at end of file

File: charts/meep-metrics-engine/templates/service.yaml [hunk 2]
```diff
@@ -14,8 +14,15 @@ spec:
...
       nodePort: {{ .Values.service.nodePort }}
-      {{- end }}
\ No newline at end of file
+      {{- end }}
+    {{- if .Values.prometheus.monitor.enabled}}
+    - name: metrics
+      port: {{ .Values.prometheus.monitor.port }}
+      targetPort: {{ .Values.prometheus.monitor.port }}
+      protocol: TCP
+    {{- end}}
```
Comment (roymx) [inline charts/meep-metrics-engine/templates/service.yaml:28]: Missing newline at end of file

File: charts/meep-mon-engine/templates/service.yaml [hunk 3]
```diff
@@ -14,8 +14,15 @@ spec:
...
       nodePort: {{ .Values.service.nodePort }}
-      {{- end }}
\ No newline at end of file
+      {{- end }}
+    {{- if .Values.prometheus.monitor.enabled}}
+    - name: metrics
+      port: {{ .Values.prometheus.monitor.port }}
+      targetPort: {{ .Values.prometheus.monitor.port }}
+      protocol: TCP
+    {{- end}}
```
Comment (roymx) [inline charts/meep-mon-engine/templates/service.yaml:28]: Missing newline at end of file

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 4]
```diff
@@ -438,35 +445,41 @@ func checkNotificationRegisteredUsers(oldZoneId string, newZoneId string, oldApI
...
+
 	resp, err := http.Post(notifyUrl, "application/json", bytes.NewBuffer(jsonNotif))
+	duration := float64(time.Since(startTime).Microseconds()) / 1000.0
 	_ = httpLog.LogTx(notifyUrl, "POST", string(jsonNotif), resp, startTime)
 	if err != nil {
 		log.Error(err)
+		met.ObserveNotification(sandboxName, serviceName, notifZonalPresence, notifyUrl, nil, duration)
 		return
 	}
+	met.ObserveNotification(sandboxName, serviceName, notifZonalPresence, notifyUrl, resp, duration)
```
Comment (roymx) [inline go-apps/meep-loc-serv/server/loc-serv.go:462]: `met.ObserveNotification(...) call on line 459 & 462 could be a single call after line 455

File: go-apps/meep-loc-serv/server/loc-serv.go [hunk 5]
```diff
@@ -438,35 +445,41 @@ func checkNotificationRegisteredUsers(oldZoneId string, newZoneId string, oldApI
...
+
 	resp, err := http.Post(notifyUrl, "application/json", bytes.NewBuffer(jsonNotif))
+	duration := float64(time.Since(startTime).Microseconds()) / 1000.0
 	_ = httpLog.LogTx(notifyUrl, "POST", string(jsonNotif), resp, startTime)
 	if err != nil {
 		log.Error(err)
+		met.ObserveNotification(sandboxName, serviceName, notifZoneStatus, notifyUrl, nil, duration)
 		return
 	}
+	met.ObserveNotification(sandboxName, serviceName, notifZoneStatus, notifyUrl, resp, duration)
```
Comment (roymx) [inline go-apps/meep-loc-serv/server/loc-serv.go:482]: Same observation as above on simplification

File: go-apps/meepctl/cmd/delete.go [hunk 6]
```diff
@@ -106,16 +106,16 @@ func deleteRun(cmd *cobra.Command, args []string) {
...
-	for _, crd := range apps {
-		cmd := exec.Command("kubectl", "delete", "crd", crd)
-		_, err := utils.ExecuteCmd(cmd, cobraCmd)
-		if err != nil {
-			err = errors.New("Error deleting CRD, name: [" + crd + "]")
-			fmt.Println(err)
-		}
-	}
-}
+// func deleteCRD(apps []string, cobraCmd *cobra.Command) {
```
Comment (roymx) [inline go-apps/meepctl/cmd/delete.go:109]: Dead code??

**Review by roymx — APPROVED** [review]
- Run license-scan as there are many new dependencies
- How did we validate if request metrics are good? (e.g. if the count is accurate)

File: charts/meep-gis-engine/templates/service.yaml (hunk 1 above)
Comment (dilallkx) [reply charts/meep-gis-engine/templates/service.yaml:28]: Fixed.

File: charts/meep-metrics-engine/templates/service.yaml (hunk 2 above)
Comment (dilallkx) [reply charts/meep-metrics-engine/templates/service.yaml:28]: Fixed.

File: charts/meep-mon-engine/templates/service.yaml (hunk 3 above)
Comment (dilallkx) [reply charts/meep-mon-engine/templates/service.yaml:28]: Fixed.

File: go-apps/meepctl/cmd/delete.go (hunk 6 above)
Comment (dilallkx) [reply go-apps/meepctl/cmd/delete.go:109]: Removed.

**Comment by dilallkx** [comment]
Ran license scan and found no significant updates.

**Comment by dilallkx** [comment]
WRT metrics accuracy, Prometheus operations that evaluate over a period of time do some extrapolation using the existing data. Therefore it is possible to seem some values that are slightly off from exact counts, especially when there are few data points. From the testing I have done the values seemed valid, however I have not verified extensively with large metric counts. I will run more tests to verify this.

**Pull request merged**